internal/engine/protocol/activitypub/data_source.go

```diff
@@ -51,90 +57,167 @@ func (s *dataSource) Start(ctx context.Context, tasksChan chan<- *engine.Tasks,
 
 	// Start consuming Kafka messages in a goroutine
 	go func() {
-		if err := retryOperation(ctx, func(ctx context.Context) error {
-			return s.consumeKafkaMessages(ctx, tasksChan)
-		}); err != nil {
-			errorChan <- err
+		if err := s.startHTTPServer(ctx); err != nil {
+			errorChan <- fmt.Errorf("start HTTP server: %w", err)
+			return
+		}
+	}()
+	// Start following relay services
+	go func() {
+		if err := s.mastodonClient.FollowRelayServices(ctx); err != nil {
+			errorChan <- fmt.Errorf("follow relay services: %w", err)
+			return
 		}
 	}()
-}
 
-// consumeKafkaMessages consumes messages from a Kafka topic and processes them
-func (s *dataSource) consumeKafkaMessages(ctx context.Context, tasksChan chan<- *engine.Tasks) error {
-	consumer := s.mastodonClient.GetKafkaConsumer()
+	go func() {
+		if err := s.processMessages(ctx, tasksChan); err != nil {
+			errorChan <- fmt.Errorf("process messages: %w", err)
+			return
+		}
+	}()
+}
 
-	// Create a pool to process messages concurrently
-	resultPool := pool.NewWithResults[*engine.Tasks]().WithMaxGoroutines(runtime.NumCPU())
+func (s *dataSource) startHTTPServer(ctx context.Context) error {
+	httpServer := echo.New()
+	httpServer.Use(middleware.Logger())
+	httpServer.Use(middleware.Recover())
 
-	// Loop to continuously process incoming messages
-	for {
-		// Start polling messages from the specified Kafka partition
-		fetches := consumer.PollFetches(ctx)
-		if errs := fetches.Errors(); len(errs) > 0 {
-			for _, e := range errs {
-				zap.L().Error("consumer poll fetch error", zap.Error(e.Err))
-			}
+	// Setup routes
+	httpServer.GET("/actor", s.handleGetActor)
+	httpServer.POST("/actor/inbox", s.handleActorInbox)
+	httpServer.POST("/inbox", s.handleInbox)
 
-			return fmt.Errorf("consumer poll fetch error: %v", errs)
+	// Start server with graceful shutdown
+	go func() {
+		if err := httpServer.Start(ServerPort); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			zap.L().Error("server error", zap.Error(err))
 		}
+	}()
```

Comment:
When the HTTP server fails to start, is it expected that the node will not stop running?

Reply:
Resolved.